spark/src/main/scala/org/apache/comet/serde/conditional.scala

```diff
@@ -91,3 +91,48 @@ object CometCaseWhen extends CometExpressionSerde[CaseWhen] {
     }
   }
 }
+
+object CometCoalesce extends CometExpressionSerde[Coalesce] {
+  override def convert(
+      expr: Coalesce,
+      inputs: Seq[Attribute],
+      binding: Boolean): Option[ExprOuterClass.Expr] = {
+    val branches = expr.children.dropRight(1).map { child =>
+      (IsNotNull(child), child)
+    }
+    val elseValue = Some(expr.children.last)
+    var allBranches: Seq[Expression] = Seq()
```

Comment:
`allBranches` seems to be redundant. It ends up being populated with the same contents as `branches`, but in a different order. It is then only used in a `withInfo` call when falling back to Spark, so the order there shouldn't matter. I think we could just use `branches` there instead.